resources/extensions/document_info_post_processor.rb

```diff
@@ -0,0 +1,38 @@
+require 'asciidoctor/extensions' unless RUBY_ENGINE == 'opal'
+
+include Asciidoctor
+
+class DocumentInfoPostProcessor < Extensions::Postprocessor; use_dsl
+  def process(document, output)
+    if document.backend == 'html5'
+      if (outfile = document.attr 'outfile')
+        require 'yaml'
+        info = {}
+        tags = []
+        if document.attr? 'tags'
```

Comment:
shouldn't it just write all document attributes to the YAML, possibly passing through a translation table for stuff that needs to be renamed? so we don't have to adjust this every time we need to handle a new attribute?

Reply:
for the other documents we also have to take care of author, neo4j-version(s) or version, category and a bunch of others.

Reply:
> shouldn't it just write all document attributes to the YAML, possibly passing through a translation table for stuff that needs to be renamed? so we don't have to adjust this every time we need to handle a new attribute?

We could but there are a lot of built-in attributes that we will need to ignore. This file is really about publishing to WordPress even though it could be a generic solution if we want to push an HTML content somewhere but we need some additional metadata.
But even then we only care about a few attributes. I guess we could use a prefix to filter the attributes? `metadata-slug`, `metadata-parent-path`? or more explicit `wordpress-slug`... 

> for the other documents we also have to take care of author, neo4j-version(s) or version, category and a bunch of others.

Are they used to publish the page to WordPress?

Reply:
They are used in WP yes.

Then perhaps it would be best to pass a positive attribute filter list + renaming (map?) to the gradle plugin to generate the yaml, not custom code that has to be adjusted.

Reply:
That's a good idea. Since it's an Asciidoctor extension we will need to configure it using a global attribute defined in the `asciidoctorj` block.

```
asciidoctorj {
    requires "${rootProject.projectDir}/resources/extensions.rb"
    attributes 'document-info-attrs-include': "slug,parent-path,tags",
}
```

I think we can automatically convert `kebab-case` to `snake_case`. The document title could also be added by default.
I'm not sure about tags but I think we could just use a `tags` attribute that can defined in an AsciiDoc document. 
So instead of:

```
= Title
:tag-a:
:tag-b:
```

We use a more explicit definition:
```
= Title
:tags: a,b
```

We could use a specific syntax to tell the extension that this attribute can contains multiple values:

```
    attributes 'document-info-attrs-include': "slug,parent-path,tags*",
```

Since `*` is not a valid character for AsciiDoc attributes it won't conflict. In this case, the extension will split (and trim) the attribute value on `,`.

Reply:
I've added an attribute called `document-metadata-attrs-include` to configure which attributes should be included.

I'm using the following suffix to tell the processor that an attribute is a list of values or a list of tuples:

**Attribute with a list of strings as value**
```
tags*
```

**Attribute with a list of tuples as value**
```
taxonomies*<>
```